### PR TITLE
Fix creep configuration handling for `undefined` and 0 differences

### PR DESCRIPTION
**Pull request by TooAngel**

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 271/456
    Coefficient: 0.594298245614
    Merging in 3 days 10 hours
    Age 0 days 15 hours

File: src/role_storagefiller.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ roles.storagefiller.killPrevious = true;
 roles.storagefiller.settings = {
   layoutString: 'MC',
   amount: [1, 4],
+  maxLayoutAmount: 1
```
Comment (mschultheiss83) [inline src/role_storagefiller.js]: maxLayoutAmount: 5

File: src/prototype_room_my.js [hunk 2]
```diff
@@ -292,7 +292,7 @@ Room.prototype.executeRoom = function() {
   const nextroomers = this.findPropertyFilter(FIND_MY_CREEPS, 'memory.role', ['nextroomer'], false, {
     filter: object => object.memory.base !== this.name
   });
-  const building = nextroomers.length > 0 && this.controller.level < 5;
+  const building = nextroomers.length > 0 && this.controller.level < 4;
```
Comment (mschultheiss83) [inline src/prototype_room_my.js]: why is that?

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 250/459
    Coefficient: 0.544662309368
    Merging in 2 days 18 hours
    Age 1 days 19 hours

File: src/role_storagefiller.js (hunk 1 above)
Comment (TooAngel) [inline src/role_storagefiller.js]: The creep will have 4 carry parts, so 200 carryCapacity, that should be enough the clean the link before the next transfer happens

File: src/prototype_room_my.js (hunk 2 above)
Comment (TooAngel) [inline src/prototype_room_my.js]: Within the last months, I improved a lot on RCL < 5. I think low rooms are now properly capable of being more effective than the `nextroomers`. I'm even thinking to reduce this more, send nextroomers only to RCL 3, but send energy to the new room.

File: src/prototype_room_basebuilder.js [hunk 3]
```diff
@@ -79,6 +79,9 @@ Room.prototype.destroyStructure = function(structure) {
         let planers = this.findPropertyFilter(FIND_MY_CREEPS, 'memory.role', ['planer']);
         if (planers.length > 3) {
           this.log('Destroying to rebuild spawn: ' + structure.structureType + ' ' + JSON.stringify(structure.pos));
+          this.log('-----------------------------------------');
+          this.log('ATTENTION: The last spawn is destroyed, a new one will be build automatically, DO NOT RESPAWN');
+          this.log('-----------------------------------------');
```
Comment (Nevrdid) [inline src/prototype_room_basebuilder.js]: may good to display it also in room visual.

File: src/prototype_room_creepbuilder.js [hunk 4]
```diff
@@ -240,7 +240,7 @@ Room.prototype.applyAmount = function(input, amount) {
   if (!input) {
     return '';
   }
-  if (typeof amount === undefined) {
+  if (amount === undefined) {
```
Comment (Nevrdid) [inline src/prototype_room_creepbuilder.js]: hum... lodash will not loop if amount not exist so i supose a simple `return output || input || ''` could allow skip input and amounts checks

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 309/462
    Coefficient: 0.668831168831
    Merging in 1 days 9 hours
    Age 1 days 22 hours

**Pull request merged**